adapters/sn2core/sn2core.go

```diff
@@ -460,46 +464,49 @@ func AdaptPreConfirmedBlock(
 		return core.PreConfirmed{}, errors.New("invalid status for pre_confirmed block")
 	}
 
-	var adaptedStateDiff *core.StateDiff
-	var err error
-
-	txStateDiffs := make([]*core.StateDiff, 0, len(response.TransactionStateDiffs))
-	for _, stateDiff := range response.TransactionStateDiffs {
-		if stateDiff == nil {
-			break
-		}
-
-		if adaptedStateDiff, err = AdaptStateDiff(stateDiff); err != nil {
-			return core.PreConfirmed{}, err
-		}
-		txStateDiffs = append(txStateDiffs, adaptedStateDiff)
+	if len(response.Transactions) != len(response.TransactionStateDiffs) || len(response.Transactions) != len(response.Receipts) {
```

Comment:
Valid linter error here. To be inline with go style please create a boolean var before as

```golang
boolVar := <condition>
if boolVar {
   ... 
}
```

This is good because the boolean var name helps the reader identify its intention and at the same time we avoid the long line or having 
```golang
  if smth || other_someht || and || so || on || lalala &&
          then || indentation || which is || not && go < friendly {
   ...
```